esx_service/utils/auth_api.py

```diff
@@ -973,24 +1021,28 @@ def _tenant_access_rm(name, datastore):
     if error_info:
         return error_info
 
-    logging.debug("_tenant_access_rm: datastore_url=%s", datastore_url)
-    error_msg = tenant.remove_datastore_access_privileges(auth_mgr.conn, datastore_url)
-    if error_msg:
-        error_info = error_code.generate_error_info(ErrorCode.INTERNAL_ERROR, error_msg)
-        return error_info
-
-    # get default_datastore, if default_datastore is the same as param "datastore"
-    # need to set default_datastore_url to "" in tenants table
+    # get dafault_datastore for this tenant
+    # if the default_datastore is equal to param "datastore", which means
+    # we are trying to remove a row in "privilege" table with datastore which is
+    # marked as default_datastore of this tenant, should return with error
     error_info, default_datastore_url = get_default_datastore_url(name)
     if error_info:
         return error_info
 
     if default_datastore_url == datastore_url:
-        error_msg = tenant.set_default_datastore(auth_mgr.conn, "")
-        if error_msg:
-            error_info = error_code.generate_error_info(ErrorCode.INTERNAL_ERROR, error_msg)
+        msg = "{} is set as default datastore".format(datastore)
+        error_info = error_code.generate_error_info(ErrorCode.PRIVILEGE_REMOVE_NOT_ALLOWED, name,
+                                                    datastore, msg)
+        return error_info
 
-    return error_info
+
+    logging.debug("_tenant_access_rm: datastore_url=%s", datastore_url)
+    error_msg = tenant.remove_datastore_access_privileges(auth_mgr.conn, datastore_url)
+    if error_msg:
+        error_info = error_code.generate_error_info(ErrorCode.INTERNAL_ERROR, error_msg)
+        return error_info
+
+    return None
```

Comment:
Can the functions that add a VM to a tenant also be checked to see if the VM can access any of the datastores (including the default) assigned to the tenant. At least warn the user that VM is added but no datastores are accessible. If default DS is not VM_DS, and the default DS is not visible on the local host for example (xESX scenario).